rholang/src/main/scala/coop/rchain/rholang/interpreter/dispatch.scala

```diff
@@ -125,14 +105,16 @@ object RholangAndScalaDispatcher {
       parallel: Parallel[M, F],
       s: Sync[M],
       ft: FunctorTell[M, Throwable]
-  ): (Dispatch[M, ListChannelWithRandom, TaggedContinuation], Reduce[M], Registry[M]) = {
+  ): (Dispatch[M, ListChannelWithRandom, TaggedContinuation], ChargingReducer[M], Registry[M]) = {
     val pureSpace          = PureRSpace[M].of(tuplespace)
     lazy val tuplespaceAlg = TuplespaceAlg.rspaceTuplespace(pureSpace, dispatcher)
     lazy val dispatcher: Dispatch[M, ListChannelWithRandom, TaggedContinuation] =
-      new RholangAndScalaDispatcher(reducer, dispatchTable)
-    lazy val reducer: Reduce[M] =
+      new RholangAndScalaDispatcher(chargingReducer, dispatchTable)
+    implicit lazy val reducer: Reduce[M] =
       new Reduce.DebruijnInterpreter[M, F](tuplespaceAlg, urnMap)
+    implicit lazy val costAlg      = CostAccountingAlg.unsafe[M](CostAccount(0))
```

Comment:
This is as bad as saying: "I'll set the field to null and initialize it using a setter later". It's also admitting we're creating this too soon, in a place we don't have all the data needed.